src/modules/45-dashboards/pages/home/DashboardForm.tsx

```diff
@@ -51,106 +50,81 @@ export interface DashboardFormikValues {
   name: string
 }
 
-function hasMatchingFolderId(folders: SelectOption[], folderId: string): boolean {
-  return !folders.some((item: SelectOption) => item.value === folderId)
-}
-
 const DashboardForm: React.FC<DashboardFormProps> = ({
+  editableFolders,
   formData,
   title,
   loading,
   onComplete,
-  modalErrorHandler,
-  setModalErrorHandler
+  setModalErrorHandler,
+  mode
 }) => {
   const { getString } = useStrings()
-  const { accountId, folderId } = useParams<DashboardPathProps>()
-
-  const {
-    data: foldersList,
-    error,
-    loading: isFoldersLoading
-  } = useGetFolders({
-    queryParams: { accountId, page: 1, pageSize: 100 }
-  })
-
-  React.useEffect(() => {
-    const errorResponse = error?.data as ErrorResponse
-    if (errorResponse?.responseMessages) {
-      modalErrorHandler?.showDanger(errorResponse.responseMessages)
-    }
-  }, [error, modalErrorHandler])
+  const { folderId } = useParams<DashboardPathProps>()
 
   const folderListItems = React.useMemo(() => {
-    if (foldersList?.resource) {
-      return foldersList.resource?.map(
-        (folder: FolderModel): SelectOption => ({ value: folder?.id, label: folder?.name })
-      )
-    }
-    return []
-  }, [foldersList])
+    return (
+      editableFolders?.map((folder: FolderModel): SelectOption => ({ value: folder?.id, label: folder?.name })) ?? []
```

Comment:
According to your props, `editableFolders` is required, so you can count on it being an array already
```suggestion
      editableFolders.map((folder: FolderModel): SelectOption => ({ value: folder?.id, label: folder?.name }))
```